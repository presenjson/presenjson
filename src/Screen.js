import React, { PureComponent } from 'react';

export default class Screen extends PureComponent {
    render() {
        const { component: Clip, ...props } = this.props;
        return (Clip && <div className={props.onScreen && 'is-on-screen'}>
            <Clip {...props} />
        </div> ||Â false);
    }
}
