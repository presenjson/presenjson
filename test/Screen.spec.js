import React from 'react';
import { shallow } from 'enzyme';
import Screen from '../src/Screen';

describe('Screen', () => {
    it('renders correctly', () => {
        const Component = () => <div />;
        expect(
            shallow(<Screen component={Component} onScreen paused={false} />)
        ).toMatchSnapshot();
    });
    it('renders nothing if no component', () => {
        expect(shallow(<Screen onScreen paused={false} />)).toMatchSnapshot();
    });
});
