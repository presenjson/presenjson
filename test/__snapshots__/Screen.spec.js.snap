// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen renders correctly 1`] = `
<div
  className="is-on-screen"
>
  <Component
    onScreen={true}
    paused={false}
    play={true}
  />
</div>
`;

exports[`Screen renders nothing if no component 1`] = `""`;
